Stage 1 frontend and api

VariusValinium22 opened the pull request:
First pull request for review and to ask to merge into the main branch

VariusValinium22 committed:
commit to this branch

VariusValinium22 committed:
Add react-router-dom to this branch also

VariusValinium22 committed:
Merge branch 'main' of https://github.com/VariusValinium22/NewsExplorer into stage-1-frontend-and-api

VariusValinium22 committed:
added the Routes/Route

VariusValinium22 committed:
added the component code.

VariusValinium22 committed:
committing addition of Saved Articles page, Sign In Modal, Sign Up Modal, Sign In/Sign Up functionality.

VariusValinium22 committed:
problem with the save icon not staying blue

VariusValinium22 committed:
finished implementing the slide-out/hamburger menu for MobileMenu

VariusValinium22 committed:
adding css

VariusValinium22 committed:
problems with the footer. Can not split the links from the icons on mobile screen

VariusValinium22 committed:
pushing to pull request for review:1

VariusValinium22 committed:
Changed File Structure in File Explorer

VariusValinium22 committed:
deployed the app to newsexplorer.flazzard.com and www.newsexplorer.flazzard.com

VariusValinium22 committed:
finished the keyword tagging and grouping for saved articles, including frequency-based sorting

VariusValinium22 committed:
all ESLint errors are resolved

ImaliaNsk commented:
Hello! Thanks for submitting your code. Your diploma project looks great!  There are a few changes needed, but I'm confident you'll handle them well. Please find comments below and make the necessary fixes. Good luck!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* The home page looks differently than in figma:
* Please check the background of the page - about and footer need white background, only the section with cards shoulf have a grey background
* The background in the header should be corrected: it's too light now, the white text is hard to read.
* The search input should be wider
* The footer layout doesn't match the design: https://prnt.sc/EvvpUbEfIa6r
* https://prnt.sc/pqjiPdtg_dyz Elements shouldn't overlap. Besides, "Search results" title and "Show more" button shouldn't be displayed when the search is active
* https://prnt.sc/LNKDHqlAtosl https://prnt.sc/2HwTVsii5Iuj saved-news page doesn't match the design. Please check the alignment of elements and background colors, and note that we don't need to display "about" section there
* Elements shouldn't touch edges: https://prnt.sc/1GraAQXlxa-z (on all resolutions)

- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.

* https://prnt.sc/MlGVF3J4IYsS Incorrectly named class

- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* "Search" button is missing an active state
* Hover effect on trash can icon from the mockup: https://prnt.sc/KFTCWRfN-fBQ

- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders, and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

* It's better to reduce the size of images. Using optimized images will save bandwidth and result in faster website loading times.

## Number of points: 79

VariusValinium22 committed:
Fixed the homepage by making the about and footer background-color: #ffffff.
Made the search input wider.
Fixed the footer layout on desktop/mobile screen sizes.
Removed 'Search results' and Show More button when Preloader is present.
Fixed Saved Articles page, got rid of about section.
Added linear-gradient to the background-image to make it darker behind the text.
Added active state to the search button for when the mouse button is held down.
added message to the delete button hover effect.
Fixed all of the className properties to proper BEM specifications.
Reduced ALL of the images size.

ImaliaNsk commented:
Fantastic work, the project looks just perfect! Congratulations, the code has passed the review. Best of luck on further stages and in your future projects!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders, and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 100
